src/library_wasmfs_opfs.js

```diff
@@ -5,70 +5,44 @@
  */
 
 mergeInto(LibraryManager.library, {
-  // TODO: Generate these ID pools from a common utility.
-  $wasmfsOPFSDirectoryHandles: {
-    allocated: [],
-    free: [],
-    get: function(i) {
+  $handleAllocator__docs: '/** @constructor */',
+  $handleAllocator: function() {
+    this.allocated = [];
+    this.freelist = [];
+    this.get = function(id) {
 #if ASSERTIONS
-      assert(this.allocated[i] !== undefined);
+      assert(this.allocated[id] !== undefined);
 #endif
-      return this.allocated[i];
-    }
-  },
-
-  $wasmfsOPFSFileHandles: {
-    allocated: [],
-    free: [],
-    get: function(i) {
-#if ASSERTIONS
-      assert(this.allocated[i] !== undefined);
-#endif
-      return this.allocated[i];
-    }
-  },
-
-  $wasmfsOPFSAccessHandles: {
-    allocated: [],
-    free: [],
-    get: function(i) {
-#if ASSERTIONS
-      assert(this.allocated[i] !== undefined);
-#endif
-      return this.allocated[i];
-    }
-  },
-
-  $wasmfsOPFSBlobs: {
-    allocated: [],
-    free: [],
-    get: function(i) {
+      return this.allocated[id];
+    };
+    this.allocate = function(handle) {
+      let id;
+      if (this.freelist.length > 0) {
+        id = this.freelist.pop();
+        this.allocated[id] = handle;
+      } else {
+        id = this.allocated.length;
+        this.allocated.push(handle);
+      }
+      return id;
+    };
+    this.free = function(id) {
 #if ASSERTIONS
-      assert(this.allocated[i] !== undefined);
+      assert(this.allocated[id] !== undefined);
 #endif
-      return this.allocated[i];
-    }
+      delete this.allocated[id];
+      this.freelist.push(id);
+    };
   },
 
-  $wasmfsOPFSAllocate: function(ids, handle) {
-    let id;
-    if (ids.free.length > 0) {
-      id = ids.free.pop();
-      ids.allocated[id] = handle;
-    } else {
-      id = ids.allocated.length;
-      ids.allocated.push(handle);
-    }
-    return id;
-  },
-
-  $wasmfsOPFSFree: function(ids, id) {
-#if ASSERTIONS
-    assert(ids.allocated[id] !== undefined);
-#endif
-    delete ids.allocated[id];
-    ids.free.push(id);
-  },
+  $wasmfsOPFSDirectoryHandles__deps: ['$handleAllocator'],
+  $wasmfsOPFSDirectoryHandles: "new handleAllocator()",
```

Comment:
Why are these strings 🤯

Reply:
Because these libraries are loaded and evaluated at compile time.   The right hand side of these expressions is what gets put into the output .. for functions this means that stringified body of the function.

If we just did `new handleAllocator()` it would try to call the `handleAllocator` function at build time.

Confusing I know..